hub/src/server/server.ts

```diff
@@ -240,7 +256,29 @@ export class HubServer {
       }
     }
 
-    const readURL = await this.driver.performWrite(writeCommand)
+    // Create a PassThrough stream to monitor streaming chunk sizes. 
+    const { monitoredStream, pipelinePromise } = monitorStreamProgress(stream, totalBytes => {
+      if (totalBytes > this.maxFileUploadSizeBytes) {
```

Comment:
This should check against the Content-Length header -- if the uploader is streaming more than the claimed Content-Length, we could close and error.

Reply:
Good catch, fixed.